### DISCUSSION TITLE
[Support]: Coral not found with Proxmox 0.14 LXC

### DISCUSSION THREAD

**Discussion by Tokn59** [blakeblackshear/frigate#13181] [General Support] [answered]
### Describe the problem you are having

I created a new instance of Frigate version 0.14 on Proxmox LXC with the exact same hardware resources as my previous install and the same config file. Everything works (except birds eye vieuw) but very very slow. Memoru allocation is 8 GB, Disk is 64 GB exactly like my older install.

### Version

0.14

### Frigate config file

```yaml
Same as previous, no errors in logfile
```

### Relevant log output

```shell
2024-08-19 11:58:57.993948  [INFO] Preparing Frigate...
2024-08-19 11:58:57.994024  [INFO] Starting Frigate...
2024-08-19 11:58:59.286896  [2024-08-19 11:58:59] frigate.app                    INFO    : Starting Frigate (0.14.0-)
2024-08-19 11:58:59.286992  [2024-08-19 11:58:59] frigate.util.config            INFO    : Checking if frigate config needs migration...
2024-08-19 11:58:59.337276  [2024-08-19 11:58:59] frigate.util.config            INFO    : frigate config does not need migration...
2024-08-19 11:58:59.444280  [2024-08-19 11:58:59] frigate.util.services          INFO    : Automatically detected vaapi hwaccel for video decoding
2024-08-19 11:58:59.479302  [2024-08-19 11:58:59] peewee_migrate.logs            INFO    : Starting migrations
2024-08-19 11:58:59.480197  [2024-08-19 11:58:59] peewee_migrate.logs            INFO    : There is nothing to migrate
2024-08-19 11:58:59.500144  [2024-08-19 11:58:59] frigate.app                    INFO    : Recording process started: 6878
2024-08-19 11:58:59.506368  [2024-08-19 11:58:59] frigate.app                    INFO    : Recording process started: 6884
2024-08-19 11:58:59.509180  [2024-08-19 11:58:59] frigate.app                    INFO    : go2rtc process pid: 126
2024-08-19 11:58:59.552004  [2024-08-19 11:58:59] frigate.app                    INFO    : Output process started: 6910
2024-08-19 11:58:59.588034  [2024-08-19 11:58:59] frigate.app                    INFO    : Camera processor started for gardenhouse: 6930
2024-08-19 11:58:59.589105  [2024-08-19 11:58:59] frigate.app                    INFO    : Camera processor started for garage: 6932
2024-08-19 11:58:59.620314  [2024-08-19 11:58:59] frigate.app                    INFO    : Camera processor started for driveway: 6935
2024-08-19 11:58:59.620520  [2024-08-19 11:58:59] frigate.app                    INFO    : Camera processor started for garden: 6941
2024-08-19 11:58:59.623785  [2024-08-19 11:58:59] frigate.app                    INFO    : Camera processor started for Livingroom: 6955
2024-08-19 11:58:59.631611  [2024-08-19 11:58:59] frigate.app                    INFO    : Capture process started for gardenhouse: 6963
2024-08-19 11:58:59.658484  [2024-08-19 11:58:59] frigate.app                    INFO    : Capture process started for garage: 6971
2024-08-19 11:58:59.659131  [2024-08-19 11:58:59] frigate.app                    INFO    : Capture process started for driveway: 6974
2024-08-19 11:58:59.676809  [2024-08-19 11:58:59] frigate.app                    INFO    : Capture process started for garden: 6980
2024-08-19 11:58:59.676875  [2024-08-19 11:58:59] frigate.app                    INFO    : Capture process started for Livingroom: 6989
```

### FFprobe output from your camera

```shell
Not relevant
```

### Frigate stats

```json
Not relevant
```

### Operating system

Debian

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Several

### Any other information that may be helpful

Could there be an obsolete entry in the config that causes this slow down ?

**Comment by hawkeye217** [comment]
We'll need to see your frigate config to be able to help.

How did you install Frigate on Proxmox? Did you use the tteck script?

**Reply by Tokn59** [reply]
Indeed installation using tteck script (advanced moe) Debian 11. I did some further testing and it looks like the slow effect only occurs when I configure my Googla Coral edgetpu. Full config : 

```
mqtt:
  host: 192.168.1.26
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: xxxxxxxxxx
  password: xxxxxxxxxxxxxxxx
  stats_interval: 60
#-------------------------------------------------------------------------------------------
cameras:
  Livingroom:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 640
      height: 320
      fps: 5
    objects:
      track:
        - person
        - cat
        - dog
        - bird
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 1
      events:
        retain:
          default: 7
#-------------------------------------------------------------------------------------------          
  garden:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxxxxxxxxx@192.168.1.169:8554/profile0
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 1920 #1280
      height: 1080  # 720
      fps: 5
    objects:
      track:
        - person
        - cat
        - dog
        - bird
        - motorcycle
        - mower
      # filters:
      #   person:
      #     min_area: 5000
      #     max_area: 100000  
      #     threshold: 0.8             
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 3
      events:
        retain:
          default: 7
#-------------------------------------------------------------------------------------------
  gardenhouse:
    ffmpeg:
      inputs:
        - path: 
            rtsp://192.168.1.238:554/user=xxxxxxx_password=xxxxxxxxx_channel=1_stream=0.sdp?real_stream
          roles:
            - detect
            - rtmp
            - record
    detect:
      # width: 640
      # height: 320
      fps: 5
      width: 1280
      height: 720

    objects:
      track:
        - person
        - cat
        - motorcycle
        - dog
        - bird
        - bicycle
      # filters:
      #   person:
      #     min_area: 5000
      #     max_area: 100000  
      #     threshold: 0.8        
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 3
      events:
        retain:
          default: 7
          mode: motion
    motion:
      mask:
        - 358,139,653,129,664,0,376,0
        - 929,720,1133,720,1123,565,925,563
#-------------------------------------------------------------------------------------------
  garage:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxx:xxxxxxxxxxxxx@192.168.1.40:554
          roles:
            - detect
            - rtmp
            - record
    detect:
      fps: 5
      width: 1280
      height: 720
    objects:
      track:
        - person
        - cat
        - dog
      filters:
        person:
          min_area: 5000
          max_area: 100000
          threshold: 0.9
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 3
      events:
        retain:
          default: 7
          mode: motion
#-------------------------------------------------------------------------------------------
  driveway:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxx:xxxxxxxxxxxxxxxx@192.168.1.42:554/cam/realmonitor?channel=1&subtype=2
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 1280
      height: 720
      fps: 5
    objects:
      track:
        - person
        - cat
        - dog
        - bird
        - motorcycle
        - bicycle
      # filters:
      #   person:
      #     min_area: 5000
      #     max_area: 100000  
      #     threshold: 0.8             
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 3
      events:
        retain:
          default: 7
#-------------------------------------------------------------------------------------------
detectors:
#  coral:
#    type: edgetpu
#    device: usb
#-------------------------------------------------------------------------------------------
  ov:
    type: openvino
    device: CPU
    model:
      path: /openvino-model/FP16/ssdlite_mobilenet_v2.xml
#-------------------------------------------------------------------------------------------      
model:
  width: 300
  height: 300
  input_tensor: nhwc
  input_pixel_format: bgr
  labelmap_path: /openvino-model/coco_91cl_bkgr.txt
version: 0.14

```

**Reply by hawkeye217** [reply]
Are you passing through the usb devices into the LXC? 

Are there any errors in Frigate's log?

**Reply by Tokn59** [reply]
The LXC contains the exavct same settings as I have in my LXC with previous Frigate version : 

`arch: amd64
cores: 4
features: nesting=1
hostname: frigate14
memory: 4096
net0: name=eth0,bridge=vmbr0,gw=192.168.1.1,hwaddr=BC:24:11:67:3F:4C,ip=192.168.1.18/24,type=veth
onboot: 1
ostype: debian
rootfs: data1:116/vm-116-disk-0.raw,size=64G
swap: 512
tags: proxmox-helper-scripts
lxc.cgroup2.devices.allow: a
lxc.cap.drop:
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyUSB1       dev/ttyUSB1       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM1       dev/ttyACM1       none bind,optional,create=file
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.apparmor.profile: unconfined
lxc.cgroup2.devices.allow: a
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=fil>
lxc.mount.entry: /dev/bus/usb/003 dev/bus/usb/003 none bind,optional,create=dir 0, 0
lxc.cap.drop:
lxc.mount.auto: cgroup:rw
`

**Reply by hawkeye217** [reply]
Are there any errors in Frigate's log when you are running with the Coral?

Can you provide a screenshot of the General and Camera pages from System metrics?

**Reply by Tokn59** [reply]
I see now that the edgetpu is not found. It was found on a previous boot. Strange ... This is the lxc config running fine withe edgetpu previous Frigate version : 
`arch: amd64
cores: 4
features: nesting=1
hostname: frigate
memory: 4096
net0: name=eth0,bridge=vmbr0,firewall=1,gw=192.168.1.1,hwaddr=16:72:DA:1E:92:E2,ip=192.168.1.18/24,type=veth
onboot: 0
ostype: debian
rootfs: data1:105/vm-105-disk-0.raw,size=128G
swap: 2048
unprivileged: 0
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.apparmor.profile: unconfined
lxc.cgroup2.devices.allow: a
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file 0, 0
lxc.mount.entry: /dev/bus/usb/003 dev/bus/usb/003 none bind,optional,create=dir 0, 0
lxc.cap.drop:
lxc.mount.auto: cgroup:rw
`

**Reply by hawkeye217** [reply]
Perhaps it's no longer at `/dev/bus/usb/003`?

**Reply by Tokn59** [reply, marked as answer]
Found the issue !!!! LXC container was running in unpriviledged mode (error in tteck scriot ?) and thus the tpu was not detected. Much better now !!!! BIG THANKS for helping me out ! Will continue testing !

**Reply by hawkeye217** [reply]
Great! Happy to help.

**Reply by Tokn59** [reply]
Maybe one more question ... The cameras are all "configured" in Go2rtc. Streams are available in the webinterface on port 1984. Do I still have to add the Go2rtc settings in the frigate config.yml ?

**Reply by hawkeye217** [reply]
> Do I still have to add the Go2rtc settings in the frigate config.yml 

Not quite sure what you mean here. What are you trying to accomplish?

What's your frigate config file?

**Reply by Tokn59** [reply]
If I remember well in previous versions(s) you had to add the Go2rtc streams in the frigate config.yml like this :

`go2rtc:
  streams:
    driveway:
     - rtsp://admin:xxxxxxxxx@192.168.1.42/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
     - "ffmpeg:driveway#video=h264"
    garage:
      - rtsp://192.168.1.51:8001/h264
      - ffmpeg:camera.garagex#audio=aac
    gardenhouse:  
      - rtsp://192.168.1.238:554/user=admin_password=xxxxxxxxxx_channel=1_stream=0.sdp?real_stream
    garden:
      - rtsp://admin:admin123456@192.168.1.169:8554/profile0/
      - "ffmpeg:garden#video=h264"

Is this still the case ?

**Reply by hawkeye217** [reply]
Yes, if you want to use restreaming. Nothing has changed from 0.13 in this regard.

**Reply by Tokn59** [reply]
Ok, THANK YOU !!!!
